src/test/java/com/jwplayer/southpaw/util/ByteArraySetTest.java

```diff
@@ -46,6 +74,651 @@ public ByteArraySet createSmallSet() {
+        List<ByteArray> vals = Stream.generate(() -> RandomStringUtils.random(RANDOM_STRING_SIZE, 0, 0, true, true, null, randomSeed))
+                .limit(count)
+                .collect(Collectors.toSet())
+                .stream()
+                .map(str -> new ByteArray(str.getBytes()))
+                .collect(Collectors.toList());
+        assertEquals(count, vals.size());
+        return vals;
+    }
+
+    private List<ByteArray> getRandomByteArrays(int count) {
+        return getRandomByteArrays(count, RANDOM_SEED);
+    }
+
+    private void testAdd(int size) {
+        System.out.println("Testing add method with " + size + " size");
+        ByteArraySet set = new ByteArraySet();
+
+        List<ByteArray> vals = getRandomByteArrays(size);
+
+        List<ByteArray> insertedVals = new ArrayList<>();
+        for (ByteArray val: vals) {
+            assertTrue(set.add(val));
+            assertTrue(set.contains(val));
+            insertedVals.add(val);
+            assertEquals(insertedVals.size(), set.size());
+        }
+
+        for (ByteArray val: vals) {
+            assertFalse(set.add(val));
+            assertTrue(set.contains(val));
+        }
+        assertEquals(size, set.size());
+    }
+
+    @Test
+    public void emptyAdd() {
+        testAdd(SIZE_EMPTY);
+    }
+
+    @Test
+    public void reallySmallAdd() {
+        testAdd(SIZE_REALLY_SMALL);
+    }
+
+    @Test
+    public void smallAdd() {
+        testAdd(SIZE_SMALL);
+    }
+
+    @Test
+    public void regularAdd() {
+        testAdd(SIZE_REGULAR);
+    }
+
+    @Test
+    public void bigAdd() {
+        testAdd(SIZE_BIG);
+    }
+
+    @Test
+    public void reallyBigAdd() {
+        testAdd(SIZE_REALLY_BIG);
+    }
+
+    private long getDifferentRandomSeed() {
+        long differentRandomSeed = RANDOM_SEED;
+        while (differentRandomSeed == RANDOM_SEED ) {
+            differentRandomSeed = new Random().nextLong();
+        }
+        return differentRandomSeed;
+    }
+
+    private void testAddAll(int size, boolean forceByteArraySet) {
+        System.out.println("Testing addAll method with " + size + " size and " + forceByteArraySet + " forceByteArraySet");
+        ByteArraySet set = new ByteArraySet();
+
+        List<ByteArray> vals = getRandomByteArrays(size);
+
+        for (ByteArray val: vals) {
+            assertTrue(set.add(val));
+        }
+        assertEquals(size, set.size());
+
+        List<ByteArray> stagedVals = getRandomByteArrays(size, getDifferentRandomSeed());
+        Set<ByteArray> extraVals;
+        if (forceByteArraySet) {
+            extraVals = new ByteArraySet();
+        } else {
+            extraVals = new HashSet<ByteArray>();
+        }
+        while(extraVals.size() < size) {
+            if (stagedVals.size() == 0) {
+                stagedVals = getRandomByteArrays(size, getDifferentRandomSeed());
+            }
+            ByteArray val = stagedVals.get(0);
+            stagedVals.remove(0);
```

Comment:
```suggestion
            ByteArray val = stagedVals.remove(0);
```
This should be equivalent to what you are doing